### PR TITLE
added support for db auto-discovery and updated base image to 8.2.0

### PR DESCRIPTION
**Pull request by deusxanima**
modified teleport-kube-agent helm chart to allow for both auto-discovery and manual db configuration and updated Chart.yaml to reflect latest available image and not 8.0.0.-alpha-1

### CODE REVIEW

**Comment by webvictim** [comment]
The `Chart.yaml` gets updated automatically on release branches by this code: https://github.com/gravitational/teleport/blob/master/version.mk#L33-L45

Us having `8.0.0-alpha.1` as the version on `master` is just a weird consequence of our not using `9.0.0-dev` or similar in the Makefile on `master`. We tried using a similar label previously, but then some deployments fail because we don't automatically package/push any `x.x.x-dev` Docker images. As such, leaving it as a version which actually has a published Docker image was preferable.

TL;DR don't update the version number in this PR, it'll just get changed back when someone runs `make version.go` anyway. It's correct on every release branch (which is where the Helm charts are now published from)

**Comment by r0mant** [comment]
@webvictim Is this PR still needed with your https://github.com/gravitational/teleport/pull/10344 PR or can we close it?

**Comment by Aharic** [comment]
@r0mant , looks like this one was mooted by @webvictim 's other PR so I'll go ahead and close it. Thanks for the feedback.
